### PR TITLE
fix frontend bug.

### PR DESCRIPTION
**Pull request by zhijxu-MS**
old ort session may already exists when creating new ort session, this may cause OOM error

### CODE REVIEW

**Comment by thiagocrepaldi** [comment]
I am on it. Will check CI and review it! Thanks!

File: orttraining/orttraining/python/training/orttrainer.py [hunk 1]
```diff
@@ -685,6 +685,9 @@ def _init_session(self):
         if self.options.utils.run_symbolic_shape_infer:
             self._onnx_model = SymbolicShapeInference.infer_shapes(self._onnx_model, auto_merge=True, guess_output_rank=True)
 
+        # old ort session may already exists and occupies GPU memory when creating new session, this may cause OOM error.
```
Comment (liqunfu) [inline orttraining/orttraining/python/training/orttrainer.py:688]: Great finding. 
NIP, shall this code be at where self._training_session is assigned?:
self._training_session = ort.TrainingSession(self._onnx_model.SerializeToString(),

**Review by liqunfu — APPROVED** [review]

File: orttraining/orttraining/python/training/orttrainer.py (hunk 1 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/orttrainer.py:688]: Agreed

Comment (zhijxu-MS) [reply orttraining/orttraining/python/training/orttrainer.py]: good point!  just resolved it.

File: orttraining/orttraining/python/ort_trainer.py [hunk 2]
```diff
@@ -696,6 +696,9 @@ def _init_session(self):
         if self.run_symbolic_shape_infer:
             self.onnx_model_ = SymbolicShapeInference.infer_shapes(self.onnx_model_, auto_merge=True, guess_output_rank=True)
 
+        # old ort session may already exists and occupies GPU memory when creating new session, this may cause OOM error.
+        # for example, load_state_dict will be called before returing the function, and it calls _init_session again
+        del self.session
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/ort_trainer.py:701]: Move this to line 461, where `TrainingSession` is really instantiated

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]

File: orttraining/orttraining/python/ort_trainer.py (hunk 2 above)
Comment (zhijxu-MS) [reply orttraining/orttraining/python/ort_trainer.py]: good point.

Comment (zhijxu-MS) [reply orttraining/orttraining/python/ort_trainer.py]: just realize that function "create_ort_training_session_with_optimizer" is not class member function, so this is the right place to delete self.session?

**Review by thiagocrepaldi — APPROVED** [review]

**Pull request merged**